app/models/piece.rb

```diff
@@ -42,8 +42,15 @@ def blocked?(x, y)
     end
 
     piece_path.each do |square|
-      return true if game.is_occupied?(square[0], square[1])
+      return true if game.square_occupied?(square[0], square[1])
     end
     false
   end
+
+  def move_to!(x, y) # move and capture
+    @game = game
+    return false unless square_occupied?(x, y)
```

Comment:
Rather than returning false, can you just wrap this in an if statement?  Something like:

```ruby
if square_occupied?(x, y)
  target_piece = game.piece.find_by(x_position: x, y_position: y)
  target_piece.update_attributes(x_position: nil, y_position: nil, captured: true)
end
```

Then that way at the end of this method, you can call `update_attributes(x_position: x, y_position: y)` to update the current piece's position after capturing the other piece, if any.